src/ultk/language/grammar.py

```diff
@@ -325,7 +325,7 @@ def add_unique(expression: GrammaticalExpression) -> None:
 
             # get lists of possible depths for each child
             for child_depths in product(range(depth), repeat=len(rule.rhs)):
-                if max(child_depths) < depth - 1:
+                if len(child_depths) == 0 or max(child_depths) < depth - 1:
```

Comment:
Can you explain the addition here and maybe add a comment in the code too?  Will this case always correspond to when `len(rule.rhs) == 0`?